src/status_im/common/controlled_input/utils.cljs

```diff
@@ -78,9 +78,10 @@
 
 (defn add-character
   [state character]
-  (when (can-add-character? state character)
+  (if (can-add-character? state character)
```

Comment:
is this all to handle the custom keyboard?
We should really raise this with the design team as something to reconsider as it seems to be a bug factory that we could easily avoid :)

Reply:
also this looks good but since this is in /common/ I wonder is this code used elsewhere and if so is this the expected behaviour for these uses?

Reply:
> is this all to handle the custom keyboard?

Yes 🤷‍♂️ 

> seems to be a bug factory 
Agreed! I think @vkjr worked on this, maybe he can share his thoughts also (and maybe @ulisesmac)

> also this looks good but since this is in /common/
It is used in the send screen only. 
The fix here is to return the same state if the character cannot be added. Previously was returning `nil` hence the crash.

Reply:
yeah let's collate of all the lists and issues this brings in. I believe it's also a big takeaway from accessibility of the end user.  @shivekkhurana was discussing something like this before.

Reply:
Hey!
I just found this discussion since was on vacation. 
Yes, if we continue to use custom keyboard, the controlled-input will be reused everywhere, this is the idea. But if in general we can reconsider using custom keyboard - I think it is even better)